plugin/pkg/admission/eventratelimit/apis/eventratelimit/v1alpha1/types.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// Type of limit (e.g., per-namespace)
+type LimitType string
+
+const (
+	// ServerLimitType limits are maintained against all events received by the server
+	ServerLimitType LimitType = "server"
+	// NamespaceLimitType limits are maintained against events from each namespace
+	NamespaceLimitType LimitType = "namespace"
+	// UserLimitType limits are maintained against events from each user
+	UserLimitType LimitType = "user"
+	// SourceObjectLimitType limits are maintained against events from each source+object
+	SourceObjectLimitType LimitType = "source+object"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Configuration provides configuration for the ResourceQuota admission controller.
+type Configuration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Limits to place on events received.
+	// Limits can be placed on events received server-wide, per namespace,
+	// per user, and per source+object.
+	// At least one limit is required.
+	Limits []Limit `json:"limits"`
+}
+
+type Limit struct {
+	// Type of limit.
+	Type LimitType `json:"type"`
```

Comment:
For public APIs (external config), the Godoc is turned into public API doc, so you should always use

a. the json field name (`type` instead of `Type`)
b. sentences that describe the use of the field.

See https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md for some examples, and also look at other config.  The docs here are pretty good, but this godoc in particular will be for end users.